packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -329,6 +335,16 @@ class PigeonOptions {
   PigeonOptions merge(PigeonOptions options) {
     return PigeonOptions.fromMap(mergeMaps(toMap(), options.toMap()));
   }
+
+  /// Returns provided or deduced package name, throws `Exception` if none found.
+  String getPackageNameOrThrow() {
+    final String? name = _packageName ?? deducePackageName(input ?? '');
+    if (name == null) {
+      throw Exception(
+          'Unable to deduce package name, and no package name supplied.');
```

Comment:
Let's explicitly add instructions about using the flag, so that someone seeing this gets clear next steps about how to fix it.